drivers/ever-hid.c

```diff
@@ -0,0 +1,638 @@
+	{ "EVER16",	0xff000014 },
+	{ "EVER17",	0xff000015 },
+	{ "EVER18",	0xff000016 },
+	{ "EVER19",	0xff000017 },
+	{ "EVER20",	0xff000018 },
+	{ "EVER21",	0xff000019 },
+	{ "EVER22",	0xff00001a },
+	{ "EVER23",	0xff00001b },
+	{ "EVER24",	0xff00001c },
+	{ "EVER25",	0xff00001d },
+	{ "EVER26",	0xff00001e },
+	{ "EVER27",	0xff00001f },
+	{ "EVER28",	0xff000020 },
+	{ "EVER29",	0xff000021 },
+	{ "EVER30",	0xff000022 },
+	{ "EVER31",	0xff000023 },
+	{ "EVER32",	0xff000030 },
+	{ "EVER33",	0xff000031 },
+	{ "EVER34",	0xff000032 },
+	{ "EVER35",	0xff000033 },
+	{ "EVER36",	0xff000034 },
+	{ "EVER37",	0xff000035 },
+	{ "EVER38",	0xff000036 },
+	{ "EVER39",	0xff000037 },
+	{ "EVER40",	0xff000038 },
+	{ "EVER41",	0xff000039 },
+	{ "EVER42",	0xff000040 },
+	{ "EVER43",	0xff000041 },
+	{ "EVER44",	0xff000042 },
+	{ "EVER45",	0xff000043 },
+	{ "EVER46",	0xff000044 },
+	{ "EVER47",	0xff000045 },
+	{ "EVER48",	0xff000046 },
+	{ "EVER49",	0xff000050 },
+	{ "EVER50",	0xff000051 },
+	{ "EVER51",	0xff000052 },
+	{ "EVER52",	0xff000053 },
+	{ "EVER53",	0xff000054 },
+	{ "EVER54",	0xff000060 },
+	{ "EVER55",	0xff000061 },
+	{ "EVER56",	0xff000062 },
+	{ "EVER57",	0xff000063 },
+	{ "EVER58",	0xff000064 },
+	{ "EVER59",	0xff000066 },
+	{ "EVER60",	0xff000067 },
+	{ "EVER61",	0xff000068 },
+	{ "EVER62",	0xff000069 },
+	{ "EVER63",	0xff00006a },
+	{ "EVER64",	0xff00006b },
+	{ "EVER65",	0xff00006c },
+	{ "EVER66",	0xff00006d },
+	{ "EVER67",	0xff00006e },
+	{ "EVER68",	0xff00006f },
+	{ "EVER69",	0xff000070 },
+	{ "EVER70",	0xff000071 },
+	{ "EVER71",	0xff000072 },
+	{ "EVER72",	0xff000073 },
+	{ "EVER73",	0xff000074 },
+	{ "EVER74",	0xff000075 },
+	{ "EVER75",	0xff000076 },
+	{ "EVER76",	0xff000077 },
+	{ "EVER77",	0xff000078 },
+	{ "EVER78",	0xff000079 },
+	{ "EVER79",	0xff00007a },
+	{ "EVER80",	0xff00007b },
+	{ "EVER81",	0xff00007c },
+	{ "EVER82",	0xff00007d },
+	{ "EVER83",	0xff00007e },
+	{ "EVER84",	0xff00007f },
+	{ "EVER85",	0xff000080 },
+	{ "EVER86",	0xff000081 },
+	{ "EVER87",	0xff000082 },
+	{ "EVER88",	0xff000083 },
+	{ "EVER89",	0xff000084 },
+	{ "EVER90",	0xff000085 },
+	{ "EVER91",	0xff000086 },
+	{ "EVER92",	0xff000087 },
+	{ "EVER93",	0xff000088 },
+	{ "EVER94",	0xff000089 },
+	{ "EVER95",	0xff00008a },
+	{ "EVER96",	0xff00008b },
+	{ "EVER97",	0xff000090 },
+	{ "EVER98",	0xff000091 },
+	{ "EVER99",	0xff000092 },
+	{ "EVER100",	0xff000093 },
+	{ "EVER101",	0xff000094 },
+	{ "EVER102",	0xff000095 },
+	{ "EVER103",	0xff000096 },
+	{ "EVER104",	0xff000097 },
+	{  NULL, 0 }
+};
+
+static usage_tables_t ever_utab[] = {
+	ever_usage_lkp,
+	hid_usage_lkp,
+	NULL,
+};
+
+
+/* --------------------------------------------------------------- */
+/* HID2NUT lookup table                                            */
+/* --------------------------------------------------------------- */
+
+static hid_info_t ever_hid2nut[] = {
+
+  { "ups.batterysystem.battery.batteryid", 0, 0, "UPS.BatterySystem.Battery.BatteryID", NULL, "%.0f", 0, NULL },
```

Comment:
The NUT names are not simply lower-case versions of the HID PDC Usage names. Please review the NUT variable guide: http://networkupstools.org/docs/developer-guide.chunked/apas01.html